### ISSUE TITLE
invalid-format-index false positive

### ISSUE THREAD

**Issue by ttsiodras** [PyCQA/pylint#2752]
The issue is NOT fixed on the preview release.

### Steps to reproduce

1. Use this file
```
    #!/usr/bin/env python2
    '''
    A bug in pylint.
    '''
    import sys

    def main():
        '''Demonstrating the bug.'''
        options = []
        if len(sys.argv) > 1:
            options = [["Woof!"]]
        else:
            options = [["Barf!"]]
        print('Why is this bad? {options[0][0]}'.format(options=options))
        print("It's clearly a bug, since pylint thinks this is fine:", options[0][0])

    if __name__ == "__main__":
        main()
```

2. Run pylint:

```
    $ pylint  bug.py  | grep -v '^$$' | sed -n '/^Report/q;p'
```

### Current behavior

```
    $ pylint  bug.py  | grep -v '^$$' | sed -n '/^Report/q;p'
    ************* Module bug
    bug.py:14:10: W1307: Using invalid lookup key 0 in format specifier 'options[0]' (invalid-format-index)
```

### Expected behavior

```
    $ pylint  bug.py  | grep -v '^$$' | sed -n '/^Report/q;p'
```

There is no bug here - and pylint reports nothing for the print command that follows, that uses exactly the same construct (`options[0][0]`).

### pylint --version output

```
    $ pylint --version output
    pylint 2.3.0-dev1
    astroid 2.2.0-dev
    Python 3.5.3 (default, Sep 27 2018, 17:25:39)
    [GCC 6.3.0 20170516]
```

But also with the current version that pip3 installs:

```
    $ pip3 install pylint astroid
    ...
    $ pylint --version output
    pylint 2.2.2
    astroid 2.1.0
    Python 3.5.3 (default, Sep 27 2018, 17:25:39)
    [GCC 6.3.0 20170516]
    
    $ pylint  bug.py  | grep -v '^$$' | sed -n '/^Report/q;p'
    ************* Module bug
    bug.py:14:10: W1307: Using invalid lookup key 0 in format specifier 'options[0]' (invalid-format-index)
```

**Comment by PCManticore** [comment]
Thanks, I can confirm the bug.
